src/sync/sync_session.cpp

```diff
@@ -347,11 +361,31 @@ void SyncSession::create_sync_session()
                 case ProtocolError::illegal_realm_path:
                 case ProtocolError::no_such_realm:
                 case ProtocolError::permission_denied:
+                case ProtocolError::bad_client_version:
+                    break;
                 case ProtocolError::bad_server_file_ident:
                 case ProtocolError::bad_client_file_ident:
                 case ProtocolError::bad_server_version:
-                case ProtocolError::bad_client_version:
-                case ProtocolError::diverging_histories:
+                case ProtocolError::diverging_histories: {
+                    // Add a SyncFileActionMetadata marking the Realm as needing to be deleted.
+                    auto full_name = new_path + to_string(std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count());
+                    error.user_info.insert({SyncError::c_original_file_path_key, path()});
+                    error.user_info.insert({SyncError::c_recovery_file_path_key, full_name});
+                    SyncManager::shared().perform_metadata_update([this,
+                                                                   old_path=this->path(),
+                                                                   user_identity=std::move(user_identity),
+                                                                   realm_url=std::move(realm_url),
+                                                                   full_name=std::move(full_name)](const SyncMetadataManager& manager) {
+                        SyncFileActionMetadata(manager,
```

Comment:
This is all drifting so far to the right that it might be a good idea to move the body of the outermost lambda into a helper function.